Initialize marsha plugins project (#1)
openfun/marsha-plugins: Readme.md

# Marsha plugins

> 

Reviewer: This README should be commit along with the Makefile from whom rules are mentioned in.

> This repo contains third party plugins we don't to commit in [marsha](https://github.com/openfun/marsha) application.

Reviewer: ```suggestion
This repo contains optional plugins for the [Marsha](https://github.com/openfun/marsha) application.
```

> plugins are pure javascript injected in the [lti template](https://github.com/openfun/marsha/blob/e5dec6b8ae662ab35950e5976296653cb29dbd7c/src/backend/marsha/core/templates/core/lti.html#L14-L16) and executed when the iframe is loaded.

Reviewer: ```suggestion
Plugins are pure JavaScript files injected in the [LTI template](https://github.com/openfun/marsha/blob/e5dec6b8ae662ab35950e5976296653cb29dbd7c/src/backend/marsha/core/templates/core/lti.html#L14-L16) and executed when the target `iframe` is loaded.
```

## Getting started

Javascript file are hosted on an AWS S3 bucket and distrbuted using AWS CloudFront. To bootstrap everything we use terraform.

🔧 **Before you go further**, you need to create `./src/aws/env.d/development` and replace the relevant values with your own.

    $ cp ./env.d/development.dist ./env.d/development

Create the shared state bucket where `Terraform` will keep all the information on your deployments so different developers/machines/CI processes can interact with them:

    $ make state-create

Initialize your `Terraform` config:

    $ make init

> In order to have multiple environment (development, staging, preproduction, production) you must create a terraform workspace and use it before deploying the project

Reviewer: ```suggestion
In order to have multiple environments (_e.g._ `development`, `staging`, `preproduction` and `production`) you must create a Terraform workspace and use it before deploying the project:
```

    $ ./bin/terraform workspace create development

> Then you can create the architecture and deploy all third party plugins

Reviewer: ```suggestion
Then you can create the architecture and deploy all plugins:
```

    $ make deploy

At then end of the deployment terraform will output the AWS CloudFront url you will use to distribute javascript files.

## Available plugins

### Adways

> Adways is an advertising interactive videos. The plugins is in `src/static/js/adways.js`. To distribute it the URL will be: `https://yourCloudFrontDomain.com/static/js/adways.js`.

Reviewer: ```suggestion
Adways is an advertising interactive videos. The plugin lives in `src/static/js/adways.js`. To distribute it the URL will be: `https://yourCloudFrontDomain.com/static/js/adways.js`.
```

> Adways is an advertising interactive videos. The plugins is in `src/static/js/adways.js`. To distribute it the URL will be: `https://yourCloudFrontDomain.com/static/js/adways.js`.

Reviewer: > Adways is an advertising interactive videos.

Should be rephrased.

Author: It took it on the adways website

Reviewer: A word is missing in this sentence. Maybe:

> Adways is an advertising videos solution.

WDYT?

Author: we should probably keep `interactive` instead of `advertising` because we use it for its interactivity

Author: Done

Reviewer: True that. I've been too fast in answering.
